### PR TITLE
Set amp source origin in requests for amp-list enabled for SSR

### PR DESCRIPTION
**Pull request by alabiaga**
b/120135852

### CODE REVIEW

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -385,6 +386,7 @@ export class AmpList extends AMP.BaseElement {
    */
   ssrTemplate_(refresh) {
     let request;
+    assertHttpsUrl(this.element.getAttribute('src'), this.element);
```
Comment (choumx) [inline extensions/amp-list/0.1/amp-list.js]: Why is this line necessary? In what cases is the URL not HTTPS?

Comment (alabiaga) [reply extensions/amp-list/0.1/amp-list.js]: the non ssr code path made the check so I added it but src is not bindable and forbidden in the amp4email spec, where if allowed would be an issue so I'll remove. Thanks

Comment (choumx) [reply extensions/amp-list/0.1/amp-list.js]: I think amp-bind also enforces that src is HTTPS-only FYI.
